spec/mongo/server/app_metadata_spec.rb

```diff
@@ -110,21 +118,10 @@
           allow(app_metadata).to receive(:driver_doc).and_return('x'*500)
         end
 
-        it 'truncates the document to be just an ismaster command and the compressors' do
-          # Because we sometimes request that the server provide a list of valid auth mechanisms for
-          # the user, we need to conditionally add the length of that metadata to the expected
-          # length of the isMaster document.
-          sasl_supported_mechs_size = 0
-          sasl_supported_mechs = app_metadata.instance_variable_get(:@request_auth_mech)
-
-          if sasl_supported_mechs
-            sasl_supported_mechs_size += 1                               # length of BSON type byte
-            sasl_supported_mechs_size += 'saslSupportedMechs'.length + 1        # length of BSON key
-            sasl_supported_mechs_size += 4                               # length of BSON string length
-            sasl_supported_mechs_size += sasl_supported_mechs.length + 1 # length of BSON string
-          end
-
-          expect(app_metadata.ismaster_bytes.length).to eq(Mongo::Server::Monitor::Connection::ISMASTER_BYTES.length + sasl_supported_mechs_size + 26)
+        it 'truncates the document' do
+          expect(
+            app_metadata.validated_document.to_bson.to_s.size
+          ).to be < described_class::MAX_DOCUMENT_SIZE
```

Comment:
I think the content of the truncation tests got largely lost with these changes. Is the validated document here static? If server versions are restricted to, say, 3.6-4.2 or 4.4+or 4.9+, is the contents then static? If I think an assertion on the contents would be helpful (that the driver_doc is omitted).